### PR TITLE
Ev2 teddy

### PR DESCRIPTION
**Pull request by teddyf**
Now when ingredient is updated, vendor will be populated with appropriate values. Inventory check now checks space vs weight, fixed issue where you couldn't see ingredient info or click on ingredient to link to ingredient page. All ingredients are in drop downs so you don't need to search for ingredient by knowing the name. Vendor now populates catalogue with actual ingredient objects rather than just searching by name or objectid.

### CODE REVIEW

File: helpers/vendor.js [hunk 1]
```diff
@@ -34,8 +44,7 @@ module.exports.makeOrder = function(ingredientId,vendorId,numUnits){
         let newName = ing['name'];
         let nativeUnit = ing['nativeUnit'];
         let unitsPerPackage = ing['unitsPerPackage'];
-        //TODO make amount on package volume
-        let amount = numUnits+parseFloat(ing['amount']);
+        let amount = numUnits*spaceMapping[package]+parseFloat(ing['amount']);
```
Comment (dy46) [inline helpers/vendor.js]: This is should be `numUnits*ing['unitsPerPackage'] + parseFloat(ing['amount'])`

I handle the inventory space stuff in the `IngredientHelper.updateIngredient` method.

File: routes/ingredients.js [hunk 2]
```diff
@@ -82,6 +83,7 @@ router.post('/:name/delete', function(req, res, next) {
       parseFloat(ing['amount'])
     );
   }).then(function() {
+    return VendorHe
```
Comment (dy46) [inline routes/ingredients.js]: I think this line is unfinished.

File: routes/vendors.js [hunk 3]
```diff
@@ -77,33 +73,37 @@ router.post('/:code/delete', function(req, res, next) {
...
+  var ingQuery = Ingredient.getIngredientById(mongoose.Types.ObjectId(req.body.ingredient));
+  ingQuery.then(async function(result) {
     if (result == null) {
       let err = new Error('This ingredient does not exist');
       return next(err);
     }
-    let ingId = mongoose.Types.ObjectId(result._id);
-    VendorHelper.addIngredient(req.params.code, ingId, req.body.cost);
-    logs.makeVendorLog('Add ingredients', { 'Vendor code': req.params.code, 'Ingredient ID': ingId, 'cost': req.body.cost }, entities = ['vendor', 'ingredient'], req.session.userId);
+    await VendorHelper.addIngredient(req.params.code, result._id, req.body.cost);
```
Comment (dy46) [inline routes/vendors.js]: Why not just do `.then` here? You return a promise after all.

File: routes/ingredients.js (hunk 2 above)
Comment (teddyf) [reply routes/ingredients.js]: my bad lol

File: routes/vendors.js (hunk 3 above)
Comment (teddyf) [reply routes/vendors.js]: i am not doing another db call so i figured if we are just waiting for a db addition before a render we didnt need the .then.   Also I kinda figured out how await works

File: helpers/vendor.js (hunk 1 above)
Comment (teddyf) [reply helpers/vendor.js]: alright. fixed this

File: routes/vendors.js (hunk 3 above)
Comment (dy46) [reply routes/vendors.js]: yeah but isn't your promise also catching and throwing errors? if you want to display those in the browser, you'll have to catch them in the route. with await you can't really do that.

Comment (teddyf) [reply routes/vendors.js]: ahhh.  I see what you mean.  Yeah i can just make it a promise i guess. tbh i was just trying to save a few lines of code lol.  Don't fully understand how promises work, just how to use them lol.  Will keep that in mind

**Pull request merged**